test/common/http/conn_pool_grid_test.cc

```diff
@@ -981,19 +980,19 @@ TEST_F(ConnectivityGridTest, RealGrid) {
   EXPECT_FALSE(grid.hasActiveConnections());
 
   // Create the HTTP/3 pool.
-  auto optional_it1 = ConnectivityGridForTest::forceCreateNextPool(grid);
-  ASSERT_TRUE(optional_it1.has_value());
-  EXPECT_EQ("HTTP/3", (**optional_it1)->protocolDescription());
+  auto pool1 = ConnectivityGridForTest::forceCreateNextPool(grid);
+  ASSERT_TRUE(pool1);
```

Comment:
nit: I think we typically use explicit nullptr checks. `ASSERT_TRUE(pool1 != nullptr)`. (Here and on line 990)